Illegible error message if sassc is missing

Running the example...
```
git clone https://github.com/cockpit-project/starter-kit.git
cd starter-kit
make
```

...ends with:
```
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! starter-kit@0.1.0 build: 'webpack'
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the starter-kit@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dev/.npm/_logs/2020-11-05T14_40_15_913Z-debug.log
make: *** [dist/index.css] Error 2
```
**2020-11-05T14_40_15_913Z-debug.log**
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build' ]
2 info using npm@6.14.8
3 info using node@v12.19.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle starter-kit@0.1.0~prebuild: starter-kit@0.1.0
6 info lifecycle starter-kit@0.1.0~build: starter-kit@0.1.0
7 verbose lifecycle starter-kit@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle starter-kit@0.1.0~build: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/dev/starter-kit/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/home/dev/bin
9 verbose lifecycle starter-kit@0.1.0~build: CWD: /home/dev/starter-kit
10 silly lifecycle starter-kit@0.1.0~build: Args: [ '-c', 'webpack' ]
11 silly lifecycle starter-kit@0.1.0~build: Returned: code: 2  signal: null
12 info lifecycle starter-kit@0.1.0~build: Failed to exec build script
13 verbose stack Error: starter-kit@0.1.0 build: `webpack`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid starter-kit@0.1.0
15 verbose cwd /home/dev/starter-kit
16 verbose Linux 3.10.0-1127.el7.x86_64
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
18 verbose node v12.19.0
19 verbose npm  v6.14.8
20 error code ELIFECYCLE
21 error errno 2
22 error starter-kit@0.1.0 build: `webpack`
22 error Exit status 2
23 error Failed at the starter-kit@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

**Versions:**
```
[root@localhost starter-kit]# node -v
v12.19.0
[root@localhost starter-kit]# npm -v
6.14.8
[root@localhost starter-kit]# webpack -v
5.4.0
```

What am I missing?

OK.
To be absolutely sure, I deployed a fresh CentOS 7 machine and tried again... but ended up in exactly the same place.

```
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! starter-kit@0.1.0 build: `webpack`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the starter-kit@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/devel/.npm/_logs/2020-11-05T15_22_00_059Z-debug.log
make: *** [dist/index.css] Error 2
```

**2020-11-05T15_22_00_059Z-debug.log**
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/bin/npm', 'run', 'build' ]
2 info using npm@6.14.8
3 info using node@v12.19.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle starter-kit@0.1.0~prebuild: starter-kit@0.1.0
6 info lifecycle starter-kit@0.1.0~build: starter-kit@0.1.0
7 verbose lifecycle starter-kit@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle starter-kit@0.1.0~build: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/devel/starter-kit/node_modules/.bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/devel/.local/bin:/home/devel/bin
9 verbose lifecycle starter-kit@0.1.0~build: CWD: /home/devel/starter-kit
10 silly lifecycle starter-kit@0.1.0~build: Args: [ '-c', 'webpack' ]
11 silly lifecycle starter-kit@0.1.0~build: Returned: code: 2  signal: null
12 info lifecycle starter-kit@0.1.0~build: Failed to exec build script
13 verbose stack Error: starter-kit@0.1.0 build: `webpack`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid starter-kit@0.1.0
15 verbose cwd /home/devel/starter-kit
16 verbose Linux 3.10.0-1127.el7.x86_64
17 verbose argv "/usr/bin/node" "/bin/npm" "run" "build"
18 verbose node v12.19.0
19 verbose npm  v6.14.8
20 error code ELIFECYCLE
21 error errno 2
22 error starter-kit@0.1.0 build: `webpack`
22 error Exit status 2
23 error Failed at the starter-kit@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

Webpack is at version 4 now.
```
[devel@localhost starter-kit]$ node -v
v12.19.0
[devel@localhost starter-kit]$ npm -v
6.14.8
[devel@localhost starter-kit]$ npm list webpack
starter-kit@0.1.0 /home/devel/starter-kit
└── webpack@4.44.2
```

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -1,4 +1,5 @@
 const path = require("path");
+const childProcess = require('child_process');
 const copy = require("copy-webpack-plugin");
 const extract = require("mini-css-extract-plugin");
 const webpack = require("webpack");
@@ -44,6 +45,18 @@ const babel_loader = {
             }],
             "@babel/preset-react"
         ]
+    }
+}
+
+/* check if sassc is available, to avoid unintelligible error messages */
+try {
+    childProcess.execFileSync('sassc', ['--version'], { stdio: ['pipe', 'inherit', 'inherit'] });
+} catch (e) {
+    if (e.code === 'ENOENT') {
+        console.error("ERROR: You need to install the 'sassc' package to build this project.");
+        process.exit(1);
+    } else {
+        throw e;
     }
 }
 

